plugins/modules/aci_l3out_dhcp_relay_label.py

```diff
@@ -0,0 +1,351 @@
+    argument_spec.update(
+        tenant=dict(type="str", aliases=["tenant_name"]),
+        l3out=dict(type="str", aliases=["l3out_name"]),
+        node_profile=dict(type="str", aliases=["node_profile_name", "logical_node"]),
+        interface_profile=dict(type="str", aliases=["interface_profile_name", "logical_interface"]),
+        dhcp_relay_label=dict(type="str", aliases=["name", "relay_policy"]),
+        scope=dict(type="str", choices=["infra", "tenant"], aliases=["owner"]),
+        dhcp_option_policy=dict(type="str", aliases=["dhcp_option_policy_name"]),
+        description=dict(type="str", aliases=["descr"]),
+        state=dict(type="str", default="present", choices=["absent", "present", "query"]),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ["state", "absent", ["tenant", "l3out", "node_profile", "interface_profile", "dhcp_relay_label"]],
+            ["state", "present", ["tenant", "l3out", "node_profile", "interface_profile", "dhcp_relay_label"]],
+        ],
+    )
+
+    tenant = module.params.get("tenant")
+    l3out = module.params.get("l3out")
+    node_profile = module.params.get("node_profile")
+    interface_profile = module.params.get("interface_profile")
+    dhcp_relay_label = module.params.get("dhcp_relay_label")
+    scope = module.params.get("scope")
+    dhcp_option_policy = module.params.get("dhcp_option_policy")
+    description = module.params.get("description")
+    state = module.params.get("state")
+
+    aci = ACIModule(module)
+
+    child_classes = ["dhcpRsDhcpOptionPol"]
+
+    aci.construct_url(
+        root_class=dict(
+            aci_class="fvTenant",
+            aci_rn="tn-{0}".format(tenant),
+            module_object=tenant,
+            target_filter={"name": tenant},
+        ),
+        subclass_1=dict(
+            aci_class="l3extOut",
+            aci_rn="out-{0}".format(l3out),
+            module_object=l3out,
+            target_filter={"name": l3out},
+        ),
+        subclass_2=dict(
+            aci_class="l3extLNodeP",
+            aci_rn="lnodep-{0}".format(node_profile),
+            module_object=node_profile,
+            target_filter={"name": node_profile},
+        ),
+        subclass_3=dict(
+            aci_class="l3extLIfP",
+            aci_rn="lifp-[{0}]".format(interface_profile),
+            module_object=interface_profile,
+            target_filter={"name": interface_profile},
+        ),
+        subclass_4=dict(
+            aci_class="dhcpLbl",
+            aci_rn="dhcplbl-[{0}]".format(dhcp_relay_label),
+            module_object=dhcp_relay_label,
+            target_filter={"name": dhcp_relay_label},
+        ),
+        child_classes=child_classes,
+    )
+
+    aci.get_existing()
+
+    if state == "present":
+        child_configs = [dict(dhcpRsDhcpOptionPol=dict(attributes=dict(tnDhcpOptionPolName=dhcp_option_policy)))]
```

Comment:
should we allow deletion of child?

Reply:
this RS relationship is not deletable. so we can't delete the child.

Reply:
UI allows deletion, in which they set the tnDhcpOptionPolName to "". Probably we should add this to the documentation.